### ISSUE TITLE
My model seems not using my custom designed loss function

### ISSUE THREAD

**Issue by Colin-Guolin** [tensorflow/tensorflow#50711]
I defined a triplet loss function as follow:

```
def triplet_loss( self, alpha ):
     def loss( y_true, y_pred ):
        anc, pos, neg = y_pred[ :, :emb_size ], y_pred[ :, emb_size:2 * emb_size ], y_pred[ :, 2 * emb_size: ]
        distance1 = tf.sqrt( tf.reduce_sum( tf.pow( anc - pos, 2 ), 1, keepdims=True ) )
        distance2 = tf.sqrt( tf.reduce_sum( tf.pow( anc - neg, 2 ), 1, keepdims=True ) )
        return tf.reduce_mean( tf.maximum( distance1 - distance2 + alpha, 0. ) )
    return loss
```

and then I compile it with my model by

   

    self.model.compile(loss=self.triplet_loss(alpha=margin,
                                              # emb_dim=emb_size
                                               ),
                        optimizer=optimizer )

then the model start training process by using `self.model.train_on_batch(data)`,
```
`Iteration 1/10000: Train loss: 4.558698, lr = 0.001000
Iteration 2/10000: Train loss: 4.558681, lr = 0.001000
Iteration 3/10000: Train loss: 4.558654, lr = 0.001000
Iteration 4/10000: Train loss: 4.558623, lr = 0.001000`
```
However, the problem is that I didn't even pass the emb_size  to the loss function. And I tried to remove the every thing in my function loss in my loss function as:
```
`def triplet_loss( self, alpha ):
    def loss( y_true, y_pred ):
        print("My loss function")
    #     anc, pos, neg = y_pred[ :, :emb_size ], y_pred[ :, emb_size:2 * emb_size ], y_pred[ :, 2 * emb_size: ]
    #     distance1 = tf.sqrt( tf.reduce_sum( tf.pow( anc - pos, 2 ), 1, keepdims=True ) )
    #     distance2 = tf.sqrt( tf.reduce_sum( tf.pow( anc - neg, 2 ), 1, keepdims=True ) )
    #     return tf.reduce_mean( tf.maximum( distance1 - distance2 + alpha, 0. ) )
    return loss`
```
and run the training program again, the training can still working, and the loss is similar to above

```
Iteration 1/10000: Train loss: 4.556189, lr = 0.001000
Iteration 2/10000: Train loss: 4.556171, lr = 0.001000
Iteration 3/10000: Train loss: 4.556145, lr = 0.001000
```

So I think the keras is not using my loss function at all, why would this happen? what should I do to fix this problem?

The complete code is [here](https://github.com/Colin-Guolin/Oneshot-Learning/blob/main/SiameseNetworkWithTripletLoss.py)

Please help me, I have struggled for a long time. Thanks in advance

**Comment by UsharaniPagadala** [comment]
@Colin-Guolin 

Could you please refer [this](https://github.com/tensorflow/tensorflow/pull/33229), and let us know if it helps.Thanks
